### PR TITLE
Send all funds on payout

### LINKED ISSUE #141: Support sending all funds from lb account on payout
As we started the discussion here https://github.com/NodeFactoryIo/vedran/pull/127#discussion_r536018061 it would be great to support default behavior when the reward amount is not defined: 

Then lb payout should use all funds on the wallet as a reward pool. Additionally, if the lb operator uses this feature, there should be a separate address to which a reward for lb itself will be sent.

### PR DESCRIPTION
**Pull request by MakMuftic**
**Enable sending the entire balance of address as reward**

### PR Checklist

- [x] I have run linter locally
- [x] I have run unit and integration tests locally
- [x] Rebased to master branch / merged master
- [x] Updated CHANGELOG.md

### Changes

- add new flag to **payout** and **start** command for address on which lb fee will be sent
- refactor payout distribution function to support separate address for loadbalancer fee
- refactor payout function to support sending entire balance of address if reward amount not defined
- _ADDITIONAL_ validate flags so it is not possible to set automatic (entire balance) reward without providing a separate address for lb fees
- _ADDITIONAL_ add prompt for the user to confirm the address for lb fees if the entire balance is going to be sent as a reward

### Issues

Closes #141

### CODE REVIEW

**Comment by MakMuftic** [comment]
@mpetrun5 you can review this code, but **we shouldn't merge this yet** as I need to do some additional manual testing to be on the safe side.

**Review by mpetrunic — COMMENTED** [review]
I'm a bit scared of this actually haha. I guess we should:
- add confirmation that user is sure that it wan't to payout everything
- maybe when starting give big fat warning if lb payout address is same as lb tresury address  or maybe don't let it start with that config?

File: internal/payout/distribution.go [hunk 1]
```diff
@@ -11,15 +11,31 @@ const (
 	requestsRewardPercentage   = 0.9
 )
 
+type LoadBalancerDistributionConfiguration struct {
+	FeePercentage float64
+	FeeAddress string
```
Comment (mpetrun5) [inline internal/payout/distribution.go]: Maybe PayoutAddress, FeeAddress this is a bit confusing

File: cmd/start.go [hunk 2]
```diff
@@ -223,22 +224,28 @@ func init() {
...
 		"",
-		"[OPTIONAL] Total reward pool in Planck",
+		"[OPTIONAL] Total reward pool in Planck. If omitted, total balance of load balancer wallet will be considered as payout reward",
 	)
 
+	startCmd.Flags().StringVar(
+		&payoutFeeAddress,
+		"lb-payout-address",
+		"",
+		"[OPTIONAL] Address on which load balancer fee will be sent. If omitted, load balancer fee will be left on load balancer wallet after payout")
```
Comment (mpetrunic) [inline cmd/start.go:241]: I think we should make it required. If you leave fee on lb, it will be distributed in next payout?

Comment (mpetrunic) [reply cmd/start.go:241]: Or make it conditional so either payout-reward or lb-payout-address is required haha

**Comment by MakMuftic** [comment]
@mpetrun5 @mpetrunic If the user wants the entire balance to be treated as reward then an additional address for lb fees must be provided. Additionally, the user is prompted to confirm this external address once more as seen in the example below
### Example
Prompt example:

image

**Comment by mpetrunic** [comment]
> @mpetrun5 @mpetrunic If the user wants the entire balance to be treated as reward then an additional address for lb fees must be provided. Additionally, the user is prompted to confirm this external address once more as seen in the example below
> 
> ### Example
> Prompt example:
> 
> image

Keep in mind we shouldn't have prompt on starting of load balancer (it would make it unusable to be run as service)

**Comment by MakMuftic** [comment]
> > @mpetrun5 @mpetrunic If the user wants the entire balance to be treated as reward then an additional address for lb fees must be provided. Additionally, the user is prompted to confirm this external address once more as seen in the example below
> > ### Example
> > Prompt example:
> > image
> 
> Keep in mind we shouldn't have prompt on starting of load balancer (it would make it unusable to be run as service)

Now prompt is only shown on the manual payout, not on starting LB

File: README.md [hunk 3]
```diff
@@ -132,7 +135,9 @@ This command has two required flags:
 
 `--private-key` - loadbalancers wallet private key (string representation of hex value prefixed with 0x), used for sending founds on payout
 
-`--payout-reward` - defined total reward amount that will be distributed on the payout (amount in Planck)
+`--payout-reward` - defined total reward amount that will be distributed on the payout (amount in Planck). If omitted entire balance of lb wallet will be used as total reward. If this options is selected than `--lb-payout-fee-address` must be set
```
Comment (mpetrunic) [inline README.md]: Wait no, if --payout-reward is omitted, --lb-payout-fee-addressmust be set

File: README.md [hunk 4]
```diff
@@ -132,7 +135,9 @@ This command has two required flags:
...
+
+`--lb-payout-fee-address` - address on which load balancer fee will be sent. If omitted, load balancer fee will be left on load balancer wallet after payout. This flag is **required** if `--payout-reward` is set to 0
```
Comment (mpetrunic) [inline README.md]: It's better to send all if -1 or not set at all?

File: README.md (hunk 3 above)
Comment (MakMuftic) [reply README.md]: That's what I meant (by **option** I meant option of sending entire balance a.k.a omitting `--payout-reward` flag) but now I see how this is very confusing wording. Fill fix

File: README.md (hunk 4 above)
Comment (MakMuftic) [reply README.md]: 0 is the default value (not set), but I will change it to -1 it makes more sense.

**Review by mpetrun5 — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**